sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala

```diff
@@ -150,8 +156,11 @@ class GpuGetJsonObjectMeta(
     lit.foreach { l =>
       val instructions = JsonPathParser.parse(l.value.asInstanceOf[UTF8String].toString)
       if (!conf.isLegacyGetJsonObjectEnabled) {
-        if (instructions.exists(JsonPathParser.fallbackCheck)) {
-          willNotWorkOnGpu("get_json_object on GPU does not support more than 16 nested paths")
+        val updated = instructions.map(JsonPathParser.filterInstructionsForJni)
+        if (updated.exists(JsonPathParser.fallbackCheck)) {
+          willNotWorkOnGpu(s"get_json_object on GPU does not support more " +
+            s"than ${JSONUtils.MAX_PATH_DEPTH} nested paths." +
+            s"(Found ${instructions.map(_.length)})")
```

Comment:
Nit: I think this isn't going to display nicely, will probably have Seq( or something on it that isn't user friendly.  Could use .mkString to tidy this up or just report the max.